### PR TITLE
Added keep_existing parameter to include_vars

### PR DESCRIPTION
**Pull request by ChaoticByte**
##### SUMMARY

I added a new parameter to include_vars named `keep_existing` (defaults to `false`). If set to `true`, existing variables won't be overwritten.  
This allow roles and playbooks to load variables/defaults more dynamically.

##### ISSUE TYPE

- Feature Pull Request
- Test Pull Request

##### ADDITIONAL INFORMATION

I added integration tests for this feature.

```sh
ansible-test integration include_vars
```

### CODE REVIEW

**Comment by ChaoticByte** [comment]
By the way, the test fails, but only after my added checks. It failed there before my change too.

File: test/integration/targets/include_vars/tasks/main.yml [hunk 1]
```diff
@@ -223,6 +223,66 @@
...
+    hash_behaviour: merge
+
+- name: Verify that the first hash variable is kept
+  assert:
+    that:
+      - "{{ config | length }} == 3"
+      - "config.key0 == 0"
+      - "config.key1 == 0"
+      - "{{ config.key2 | length }} == 1"
+      - "config.key2.a == 21"
```
Comment (flowerysong) [inline test/integration/targets/include_vars/tasks/main.yml]: Don't nest moustaches. `that` takes a list of Jinja expressions, so it should not contain additional Jinja delimiters.
```suggestion
      - config | length == 3
      - config.key0 == 0
      - config.key1 == 0
      - config.key2 | length == 1
      - config.key2.a == 21
```

Comment (ChaoticByte) [reply test/integration/targets/include_vars/tasks/main.yml]: Makes sense.
I corrected that, also for the other asserts along the way.

**Comment by ChaoticByte** [comment]
@webknjaz @flowerysong 
I took another look at why the integration tests didn't run, and discovered that I messed up the last one by putting my vars files into the wrong directory. This is now fixed :)

**Comment by nitzmahone** [comment]
The existing mess around `hash_behaviour` that we're still :thinking: on how to clean up would likely be further complicated by trying to implement and support something like this.

If you need this behavior, it's probably best to just implement a local version of include_vars in the meantime- we're unlikely to accept this change in the core version anytime soon.

**Pull request closed without merging**